lib/dynflow/action/with_bulk_sub_plans.rb

```diff
@@ -0,0 +1,83 @@
+module Dynflow
+  module Action::WithBulkSubPlans
+    include Dynflow::Action::Cancellable
+
+    BATCH_SIZE = 10
+
+    # Should return a slice of size items starting from item with index from
+    def batch(from, size)
+      raise NotImplementedError
+    end
+
+    PlanNextBatch = Algebrick.atom
+
+    def run(event = nil)
+      if event === PlanNextBatch
+        spawn_plans if can_spawn_next_batch?
+        suspend
+      else
+        super
+      end
+    end
+
+    # Should return the expected total count of tasks
+    def total_count
+      raise NotImplementedError
+    end
+
+    # Returns the items in the current batch
+    def current_batch
+      start_position = output[:total_count]
```

Comment:
I suggest keeping the total_count in output to match the `total_count` method, and introduce `planned_count` to keep the number of tasks we have planned: The `total_count` is misleading.